pyvista/core/utilities/features.py

```diff
@@ -13,8 +13,11 @@
 from .helpers import wrap
 
 
-def voxelize(mesh, density=None, check_surface=True):
-    """Voxelize mesh to UnstructuredGrid.
+def voxelize(mesh, density=None, vol=False, check_surface=True):
+    """Voxelize mesh to create an UnstructuredGrid or a StructuredGrid.
+
+       1. Voxelize the input mesh : UnstructuredGrid.
+       2. Create a voxel volume that encloses the input mesh : StructuredGrid.
```

Comment:
Both modes here will not give meshes with `vtkVoxel` cells, but rather `vtkHexahedron` cells.  Why aren't we using `RectilinearGrid` for these? If we use the vtk blanking for this new mode, I think we need `StructuredGrid`.  If we don't, I think we can use `RectilinearGrid` to get true voxels.

Reply:
This is a good point @MatthewFlamm. I've rewritten the function that facilitates generation a voxel volume. I've added an `if-else` statement that creates a `StructuredGrid` if the input `density` values **are** equal, and a `RectilinearGrid` if the `density` values are _not_ equal.